### PR TITLE
Include sems data in tally reports

### PR DESCRIPTION
**Pull request by carolinemodic**
Includes sems file in the ballot count tables and tally reports. When there is a SEMs file loaded, adds a "ballots cast by data source" table to the tally report screen page, and the tally reports themselves. When there is not a SEMs file loaded this is just 1 line with the total number of ballots. I imagine adding another table here with the ballot breakdown by voting method eventually as well. 
Tally Screen:
Screen Shot 2021-02-17 at 10 07 28 AM

Tally Report Screen:
(If there are multiple reports in this tally report, i.e. by party or for all precincts this table is for all the ballots combined across all reports)
Screen Shot 2021-02-17 at 9 31 19 AM

Tally Report:
Screen Shot 2021-02-17 at 9 35 21 AM

### CODE REVIEW

File: apps/election-manager/src/utils/semsTallies.ts [hunk 1]
```diff
@@ -333,3 +340,53 @@ export function getPrecinctIdsInExternalTally(
...
+
+  // Filter contests by party and recompute the number of ballots based on those contests.
+  const districtsForParty = getDistrictIdsForPartyId(election, partyId)
+  const filteredContestTallies: Dictionary<ContestTally> = {}
+  Object.keys(filteredTally.contestTallies).forEach((contestId) => {
+    const contestTally = filteredTally.contestTallies[contestId]
+    if (
+      !!contestTally &&
+      districtsForParty.includes(contestTally?.contest.districtId) &&
+      contestTally?.contest.partyId === partyId
```
Comment (eventualbuddha) [inline apps/election-manager/src/utils/semsTallies.ts:379]:
```suggestion
      contestTally &&
      districtsForParty.includes(contestTally.contest.districtId) &&
      contestTally.contest.partyId === partyId
```

Unless you need a `boolean`, using `!!` is superfluous. Also, TS can infer that `contestTally` is not `undefined` which makes the optional chaining unnecessary (and confusing).

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 2]
```diff
@@ -36,18 +39,63 @@ const TallyHeader = styled.div`
...
+              <TD>VotingWorks Data</TD>{' '}
+              <TD textAlign="right">{format.count(votingWorksBallotCount)}</TD>
+            </tr>
+            <tr>
+              <TD>Imported SEMs File</TD>{' '}
+              <TD textAlign="right">{format.count(externalBallotCount)}</TD>
+            </tr>
+            <tr>
+              <TD>Total</TD>{' '}
+              <TD textAlign="right">{format.count(totalBallotCount)}</TD>
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:89]:
```suggestion
              <TD><strong>Total</strong></TD>
              <TD textAlign="right"><strong>{format.count(totalBallotCount)}</strong></TD>
```

**Review by eventualbuddha — APPROVED** [review]

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 3]
```diff
@@ -36,18 +39,63 @@ const TallyHeader = styled.div`
...
+              <TD as="th" textAlign="right">
+                Number of Ballots Cast
+              </TD>
+            </tr>
+            <tr>
+              <TD>VotingWorks Data</TD>{' '}
+              <TD textAlign="right">{format.count(votingWorksBallotCount)}</TD>
+            </tr>
+            <tr>
+              <TD>Imported SEMs File</TD>{' '}
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:84]:
```suggestion
              <TD>Imported SEMs File</TD>
```

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 4]
```diff
@@ -36,18 +39,63 @@ const TallyHeader = styled.div`
...
+        <Table>
+          <tbody>
+            <tr>
+              <TD as="th">Data Source</TD>{' '}
+              <TD as="th" textAlign="right">
+                Number of Ballots Cast
+              </TD>
+            </tr>
+            <tr>
+              <TD>VotingWorks Data</TD>{' '}
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:80]:
```suggestion
              <TD>VotingWorks Data</TD>
```

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 5]
```diff
@@ -36,18 +39,63 @@ const TallyHeader = styled.div`
...
+      ? votingWorksBallotCount
+      : votingWorksBallotCount + externalBallotCount
+  const ballotsByDataSource =
+    externalBallotCount !== undefined ? (
+      <MetadataSection>
+        <h3>Ballots Cast by Data Source</h3>
+        <Table>
+          <tbody>
+            <tr>
+              <TD as="th">Data Source</TD>{' '}
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:74]:
```suggestion
              <TD as="th">Data Source</TD>
```

File: apps/election-manager/src/screens/TallyScreen.tsx [hunk 6]
```diff
@@ -209,6 +223,13 @@ const TallyScreen: React.FC = () => {
         </tbody>
       </Table>
       <h2>Ballot Count by Scanner</h2>
+      {fullElectionExternalTally && (
+        <p>
+          The following results only include ballots counted with VotingWorks
+          scanners. The data from the imported SEMs file is not included in
+          these reports.
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyScreen.tsx:230]: SEMS not SEMs

```suggestion
          The following results only include ballots counted with VotingWorks
          scanners. The data from the imported SEMS file is not included in
          these reports.
```

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 7]
```diff
@@ -36,18 +39,63 @@ const TallyHeader = styled.div`
   }
 `
 
+const MetadataSection = styled.div`
+  h3 {
+    margin-bottom: 0.25em;
+  }
+`
+
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:47]: This is not necessary if you replace the `<div>` with a `<React.Fragment>`. Prose will handle all spacing between typographical elements.

**Review by beausmith — APPROVED** [review]
LGTM with suggested changes and this one small requested "related" change 😉 

Can you change:

image

to be `<TD narrow nowrap>`

**Pull request merged**